fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PruneAggChildColumns.java

```diff
@@ -54,15 +59,63 @@ public class PruneAggChildColumns extends OneRewriteRuleFactory {
     @Override
     public Rule build() {
         return RuleType.COLUMN_PRUNE_AGGREGATION_CHILD.build(logicalAggregate().then(agg -> {
+            Slot slot = handleAggregateConstant(agg);
+            List<Slot> childOutput = agg.child().getOutput();
+            if (slot != null) {
+                if (childOutput.size() == 1 && childOutput.get(0).equals(slot)) {
+                    return agg;
+                }
+                return agg.withChildren(ImmutableList.of(new LogicalProject<>(ImmutableList.of(slot), agg.child())));
```

Comment:
it is wrong that use new slot to replace child's output, to consider this situation:
```sql
select count(1) from t group by a
```
the child's output is just `a`. we cannot replace it with new slot.